api/client/create.go

```diff
@@ -24,25 +24,41 @@ import (
 
 // Create is responsible for pushing a ToDD object to the server for eventual storage in whatever database is being used
 // It will send a ToddObject rendered as JSON to the "createobject" method of the ToDD API
-func (capi ClientApi) Create(conf map[string]string, objFile string) {
+func (capi ClientApi) Create(conf map[string]string, yamlFileName string) {
 
-	// If no subarg was provided, do nothing special
-	if objFile == "" {
-		fmt.Println("Please provide definition file")
-		os.Exit(1)
+	// See if anything is being passed via stdin
+	fi, err := os.Stdin.Stat()
+	if err != nil {
+		panic(err)
 	}
 
-	// Read YAML file
-	filename, _ := filepath.Abs(fmt.Sprintf("./%s", objFile))
-	yamlFile, err := ioutil.ReadFile(filename)
-	if err != nil {
-		fmt.Println("Unable to parse YAML")
-		os.Exit(1)
+	var yamlDef []byte
+	// If stdin is populated, read from that
+	if fi.Size() > 0 {
+		yamlDef, err = ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			panic(err)
+		}
+	} else {
+
+		// Quit if there's nothing on stdin, and there's no arg either
+		if yamlFileName == "" {
+			fmt.Println("Please provide definition file")
+			os.Exit(1)
+		}
+
+		// Read YAML file
+		filename, _ := filepath.Abs(fmt.Sprintf("./%s", yamlFileName))
```

Comment:
Not really. I think I originally meant to test how this would work if I didn't specify the ```./``` but never did. Will remove and test - definitely a cleaner way.